t/plugin/request-id2.t

```diff
@@ -0,0 +1,184 @@
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "request-id": {
+                                "algorithm": "range_id"
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1982": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/opentracing"
+                }]]
+                )
+            if code >= 300 then
+                ngx.status = code
+            end
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 3: add plugin with algorithm range_id
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local http = require "resty.http"
+            local v = {}
+            local ids = {}
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                       "plugins": {
+                            "request-id": {
+                                "algorithm": "range_id",
+                                "range_id": {}
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1982": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/opentracing"
+                }]]
+                )
+            if code >= 300 then
+                ngx.say("algorithm range_id is error")
+            end
+            for i = 1, 180 do
+                local th = assert(ngx.thread.spawn(function()
+                    local httpc = http.new()
+                    local uri = "http://127.0.0.1:" .. ngx.var.server_port .. "/opentracing"
+                    local res, err = httpc:request_uri(uri,
+                        {
+                            method = "GET",
+                            headers = {
+                                ["Content-Type"] = "application/json",
+                            }
+                        }
+                    )
+                    if not res then
+                        ngx.log(ngx.ERR, err)
+                        return
+                    end
+                    local id = res.headers["X-Request-Id"]
+                    if not id then
+                        return -- ignore if the data is not synced yet.
+                    end
+                    if #id ~= 16 then
+                        ngx.say(id)
+                        ngx.say("incorrect length for id")
+                        return
+                    end
+                    for j = 180, #id do
```

Comment:
The content check code is confusing. Could you explain how it works?

Reply:
`for j = 1, #id do` The var `j` should start from 1. At first, i want use `[a-zA-Z0-9]{16}` regex expression to check the id if correct, but Lua seems don't support `{}` syntax， so i use for loop to check each  character.

Reply:
But this may be a stupid way, if you have better one, I will improve it.

Reply:
There is a huge improvement space for the current O(n^2) solution. Please think about a better one.

Reply:
I updated the test.